rocAL/rocAL/source/augmentations/geometry_augmentations/node_crop_mirror_normalize.cpp

```diff
@@ -26,59 +26,92 @@ THE SOFTWARE.
 #include "exception.h"
 
 
-CropMirrorNormalizeNode::CropMirrorNormalizeNode(const std::vector<Image *> &inputs,
-                                                 const std::vector<Image *> &outputs) :
+CropMirrorNormalizeNode::CropMirrorNormalizeNode(const std::vector<rocalTensor *> &inputs,
+                                                 const std::vector<rocalTensor *> &outputs) :
         Node(inputs, outputs),
-        _mirror(MIRROR_RANGE[0], MIRROR_RANGE[1])
-{
+        _mirror(MIRROR_RANGE[0], MIRROR_RANGE[1]) {
         _crop_param = std::make_shared<RocalCropParam>(_batch_size);
 }
 
-void CropMirrorNormalizeNode::create_node()
-{
+void CropMirrorNormalizeNode::create_node() {
     if(_node)
         return;
 
     if(_crop_param->crop_h == 0 || _crop_param->crop_w == 0)
         THROW("Uninitialized destination dimension - Invalid Crop Sizes")
 
     _crop_param->create_array(_graph);
-    _mean_vx.resize(_batch_size);
-    _std_dev_vx.resize(_batch_size);
-    for (uint i=0; i < _batch_size; i++ ) {
-         _mean_vx[i] = _mean;
-         _std_dev_vx[i] = _std_dev;
+
+    std::vector<float> mean_vx, std_dev_vx;
+    int mean_stddev_array_size = _batch_size * _inputs[0]->info().get_channels();
+    if(!_std_dev[0])
+        THROW("Standard deviation value cannot be 0");
+    mean_vx.resize(mean_stddev_array_size, -(_mean[0] / _std_dev[0]));
+    std_dev_vx.resize(mean_stddev_array_size, (1 / _std_dev[0]));
+    
+    if(_inputs[0]->info().get_channels() == 3) {
```

Comment:
Have we handled the case when get_channels() != 3

Reply:
We are initializing the vector values with the same mean and std dev values, and if channels = 3 we are changing the values of each element in vector differetly